File: app/src/main/java/com/weeknday/onething/GlobalApplication.java
package com.weeknday.onething;
import android.app.Activity;
import android.app.Application;
import android.util.Log;

import com.kakao.auth.KakaoSDK;


public class GlobalApplication extends Application {
    private static GlobalApplication mInstance;

    public static GlobalApplication getGlobalApplicationContext() {
        if(mInstance == null)
            throw new IllegalStateException("this application does not inherit GlobalApplication");
        return mInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;

        KakaoSDK.init(new KakaoSDKAdapter());
    }
}

File: app/src/main/java/com/weeknday/onething/GlobalApplication.kt
package com.weeknday.onething

import android.app.Application
import com.kakao.auth.KakaoSDK


class GlobalApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        mInstance = this
        KakaoSDK.init(KakaoSDKAdapter())
    }

    companion object {
        private var mInstance: GlobalApplication? = null
        val globalApplicationContext: GlobalApplication?
            get() {
                checkNotNull(mInstance) { "this application does not inherit GlobalApplication" }
                return mInstance
            }
    }
}
